dist/modules/checkboxFilterGroup/checkboxFilterGroup.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+var moduleName = 'c2fo.ui.modules.checkboxFilterGroup';
+
+angular.module(moduleName, [])
+    .directive('c2foCheckboxFilterGroup', [function () {
+        return {
+            restrict: 'E',
+            scope: {
+                label: '@',
+                fields: '=',
+                filter: '=',
+                filterKey: '@',
+                multicheck: '=',
+                onUpdate: '&'
+            },
+            template: [
+                '<div class="c2fo-ui-checkbox-filter-group">',
+                '   <div class="heading">{{checkBoxFilterGroupCtrl.label}}</div>',
+                '   <div class="checkbox" ng-repeat="field in checkBoxFilterGroupCtrl.fields">',
+                '       <label>',
+                '           <input type="checkbox" ng-model="field.isChecked" ng-click="checkBoxFilterGroupCtrl.updateFilter($event)"/>',
+                '           {{field.label}}',
+                '       </label>',
+                '   </div>',
+                '</div>'
+            ].join('').replace(/\s\s+/g, ''),
+            controller: 'CheckboxFilterGroupController',
+            controllerAs: 'checkBoxFilterGroupCtrl',
+            bindToController: true
+        };
+    }])
+    .controller('CheckboxFilterGroupController', ['$timeout', function ($timeout) {
+        var vm = this;
+
+        vm.updateFilter = updateFilter;
+
+        function updateFilter(event) {
+            if (vm.multicheck) {
+                var values = [];
+
+                vm.fields.forEach(function (field) {
+                    if (field.isChecked === true) {
+                        values.push(field.value);
+                    }
+                });
+                vm.filter[vm.filterKey] = values;
```

Comment:
It would be very slightly cleaner if we use the newer array operators (`filter` and `map`) here instead of manually creating the `values` array.

```javascript
vm.filter[vm.filterKey] = vm.fields
    .filter(function (field) {
        return field.isChecked;
    }).map(function(filed) {
        return filed.value;
    });
```